packages/blade/src/components/Button/IconButton/IconButton.tsx

```diff
@@ -31,25 +31,46 @@ type IconButtonProps = {
    * Sets aria-label to help users know what the action does, eg 'Dismiss alert'
    */
   accessibilityLabel: string;
-};
+} & TooltipTriggerProps;
 
 /**
  * Component for making clickable icons with transparent background.
  * For other cases please use `Button` component with `icon` prop.
  */
 const _IconButton: React.ForwardRefRenderFunction<BladeElementRef, IconButtonProps> = (
-  { icon, onClick, size = 'medium', contrast = 'low', accessibilityLabel },
+  {
+    icon,
+    onClick,
+    size = 'medium',
+    contrast = 'low',
+    accessibilityLabel,
+    onBlur,
+    onFocus,
+    onMouseLeave,
+    onMouseMove,
+    onPointerDown,
+    onPointerEnter,
+    onTouchEnd,
+    onTouchStart,
+  },
   ref,
 ) => {
-  const innerRef = useBladeInnerRef(ref);
   return (
     <StyledIconButton
-      ref={innerRef as any}
+      ref={ref as any}
       onClick={onClick}
       contrast={contrast}
       size={size}
       icon={icon}
       accessibilityLabel={accessibilityLabel}
+      onBlur={onBlur}
+      onFocus={onFocus}
+      onMouseLeave={onMouseLeave}
+      onMouseMove={onMouseMove}
+      onPointerDown={onPointerDown}
+      onPointerEnter={onPointerEnter}
+      onTouchEnd={onTouchEnd}
+      onTouchStart={onTouchStart}
```

Comment:
Maybe have a props map that we can perhaps do `{ ...triggerProps }`

Reply:
Would work, but then in the components we will have to accept 

```jsx
<Button triggerProps={} /> 
```

Not sure if this is going to be better than just accepting all the trigger props in the Button as is, like any other react component. 🤔